### PR TITLE
test_tour_test_survey_form_triggers

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/survey/static/src/question_page/question_page_one2many_field.js [hunk 1]
```diff
@@ -48,10 +48,7 @@ class QuestionPageOneToManyField extends X2ManyField {
         const saveRecord = async (record) => {
             let isSaved = await superSaveRecord(record);
```
Comment (aab-odoo) [inline addons/survey/static/src/question_page/question_page_one2many_field.js]: undefined if I remember correctly, isn't there something to do about that?

File: addons/web/static/src/views/relational_model/static_list.js [hunk 2]
```diff
@@ -875,7 +875,7 @@ export class StaticList extends DataPoint {
         await Promise.all(proms);
 
         await this._sort();
-        this._onUpdate();
+        await this._onUpdate();
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/static_list.js:878]: qunit for this? resequence, block onchange on main record, save -> should not wait for onchange to return to save

File: addons/survey/static/src/question_page/question_page_one2many_field.js (hunk 1 above)
Comment (jpp-odoo) [reply addons/survey/static/src/question_page/question_page_one2many_field.js]: validateExtendedRecord return undefined.
but replaceWith return a promise...

**Review by aab-odoo — APPROVED** [review]

File: addons/survey/static/src/question_page/question_page_one2many_field.js [hunk 3]
```diff
@@ -27,6 +27,30 @@ patch(X2ManyFieldDialog.prototype, 'survey_question_chaining_with_validation', {
     }
 });
 
+
+/**
+ * For convenience, we'll prevent closing the question form dialog and
+ * stay in edit mode to make sure only valid records are saved. Therefore,
+ * two things should be cared for in case of error occurring when saving
+ * the question:
+ *   * Remove erroneous question row added to the embedded list
```
Comment (aab-odoo) [inline addons/survey/static/src/question_page/question_page_one2many_field.js]: not relevant here

File: addons/survey/static/src/question_page/question_page_one2many_field.js [hunk 4]
```diff
@@ -46,30 +70,22 @@ class QuestionPageOneToManyField extends X2ManyField {
...
-                const isSucceed = await self.props.record.save({
-                    stayInEdition: true,
-                    throwOnError: true,
-                });
-                isSaved = isSaved === false ? false : isSucceed;
+                await self.props.record.save();
             } catch (error) {
-                self.handleSurveySaveError(error, record);
-                return false;
+                await this.list.delete(record);
```
Comment (aab-odoo) [inline addons/survey/static/src/question_page/question_page_one2many_field.js:77]: rather mention it here

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**